### ISSUE TITLE
http_client test failing on mac with a segfault.

### ISSUE THREAD

**Issue by Lord-Kamina** [Microsoft/cpprestsdk#903]
Making a macports portfile, on 2.10.6, testing has following output:

> Running tests...
/opt/local/bin/ctest --force-new-ctest-process 
Test project /opt/local/var/macports/build/_opt_portfiles_www_cpprestsdk/cpprestsdk/work/build
    Start 1: httpclient_test
1/7 Test #1: httpclient_test ..................***Exception: SegFault 29.16 sec

All other tests pass.

Running those tests manually via test_runner, I get this:

> Test case request_helper_tests:overwrite_user_agent PASSED
Starting test case request_stream_tests:set_body_stream_1...
/opt/local/var/macports/build/_opt_portfiles_www_cpprestsdk/cpprestsdk/work/cpprestsdk-2.10.6/Release/tests/functional/http/client/request_stream_tests.cpp:96: error: Failure in set_body_stream_1: Unhandled exception: No such file or directory: No such file or directory FAILED
Test case request_stream_tests:set_body_stream_1 FAILED
Starting test case request_stream_tests:set_body_stream_2...
/opt/local/var/macports/build/_opt_portfiles_www_cpprestsdk/cpprestsdk/work/cpprestsdk-2.10.6/Release/tests/functional/http/client/request_stream_tests.cpp:123: error: Failure in set_body_stream_2: Unhandled exception: No such file or directory: No such file or directory FAILED
Test case request_stream_tests:set_body_stream_2 FAILED
Starting test case request_stream_tests:without_content_length_1...
/opt/local/var/macports/build/_opt_portfiles_www_cpprestsdk/cpprestsdk/work/cpprestsdk-2.10.6/Release/tests/functional/http/client/request_stream_tests.cpp:193: error: Failure in without_content_length_1: Unhandled exception: No such file or directory: No such file or directory FAILED
libc++abi.dylib: terminating
Trace/BPT trap: 5

**Comment by BillyONeal** [comment]
Does it still repro for you with master? I just merged a fix for a bunch of flaky test problems.

**Comment by Lord-Kamina** [comment]
I will check tonight but I'd say probably yes, from a quick look at your commit, it doesn't seem to touch the files mentioned in the log I posted.

**Comment by BillyONeal** [comment]
Right, that's because there were product bugs; the tests were catching legitimate races and memory corruption bugs in cpprestsdk proper.

**Comment by Lord-Kamina** [comment]
Will try them again tonight when I get off work then.

**Comment by Lord-Kamina** [comment]
95409c5c72a26f6a587ace70741fd2bce5dab7ec is passing all tests now on macOS 10.14
